src/script/lua_api/l_item.cpp

```diff
@@ -612,9 +612,12 @@ int ModApiItemMod::l_get_content_id(lua_State *L)
 	std::string name = luaL_checkstring(L, 1);
 
 	const NodeDefManager *ndef = getGameDef(L)->getNodeDefManager();
-	content_t c = ndef->getId(name);
+	IItemDefManager *idef = getGameDef(L)->getItemDefManager();
+	content_t content_id;
+	if (!ndef->getId(idef->getAlias(name), content_id))
+		throw LuaError("Unknown node: " + name);
```

Comment:
Lua is written in C and so doesn't support exceptions. Use lua_error instead

Reply:
`throw` is used in many other places, e.g. vmanip:
https://github.com/minetest/minetest/blob/580e7e8eb902ae2faed36b4982e7e751e35f5201/src/script/lua_api/l_vmanip.cpp#L86-L94

Reply:
I'm not sure why this works with plain Lua, but for LuaJIT there's something called `script_exception_wrapper` which catches exceptions and translations to `lua_error`. I guess plain Lua is fine because it is embedded?